File: python/examples/paths.py


def combine(abs_path, paths):
    cds = paths.split("/")
    pwd = abs_path.split("/")

    for cd in cds: 
        if(len(pwd) > 1 and cd == ".."):
            pwd.pop(len(pwd) - 1)
        elif(cd == "."):
            pass
        else:
            pwd.append(cd)

    return "/".join(pwd)


print(combine("test/foo/test", "hello/jlfk/magic/../"))

File: js/paths.js


function combin(abs_path, path){
    cds = path.split("/");
    pwd = abs_path.split("/");

    for(cd in cds){
        if(pwd.length > 1 && cds[cd] == ".."){
            pwd.pop();
        }else if(cds[cd]== "."){
            continue;
        }else{
            pwd.push(cds[cd]);
        }
    }
    
    return pwd.join("/");

}

console.log(combin("test/sdfs/test", "ete/setst/../."));
